src/Gui/ViewProviderDragger.cpp

```diff
@@ -81,249 +115,228 @@ bool ViewProviderDragger::doubleClicked()
 
+    csysDragger->addStartCallback(dragStartCallback, this);
     csysDragger->addFinishCallback(dragFinishCallback, this);
+    csysDragger->addMotionCallback(dragMotionCallback, this);
 
-    // dragger node is added to viewer's editing root in setEditViewer
-    // pcRoot->insertChild(csysDragger, 0);
-    csysDragger->ref();
+    Gui::Control().showDialog(getTransformDialog());
 
-    auto task = new TaskCSysDragger(this, csysDragger);
-    Gui::Control().showDialog(task);
-  }
+    updateDraggerPosition();
 
-  return true;
+    return true;
 }
 
 void ViewProviderDragger::unsetEdit(int ModNum)
 {
-  Q_UNUSED(ModNum);
-
-  if(csysDragger)
-  {
-    pcTransform->translation.disconnect(&csysDragger->translation);
-    pcTransform->rotation.disconnect(&csysDragger->rotation);
-
-    // dragger node is added to viewer's editing root in setEditViewer
-    // pcRoot->removeChild(csysDragger); //should delete csysDragger
-    csysDragger->unref();
-    csysDragger = nullptr;
-  }
-  Gui::Control().closeDialog();
+    Q_UNUSED(ModNum);
+
+    csysDragger.reset();
+
+    Gui::Control().closeDialog();
 }
 
 void ViewProviderDragger::setEditViewer(Gui::View3DInventorViewer* viewer, int ModNum)
 {
     Q_UNUSED(ModNum);
 
-    if (csysDragger && viewer)
-    {
-      auto rootPickStyle = new SoPickStyle();
-      rootPickStyle->style = SoPickStyle::UNPICKABLE;
-      auto selection = static_cast<SoGroup*>(viewer->getSceneGraph());
-      selection->insertChild(rootPickStyle, 0);
-      viewer->setSelectionEnabled(false);
-      csysDragger->setUpAutoScale(viewer->getSoRenderManager()->getCamera());
-
-      auto mat = viewer->getDocument()->getEditingTransform();
-      viewer->getDocument()->setEditingTransform(mat);
-      auto feat = dynamic_cast<App::GeoFeature *>(getObject());
-      if(feat) {
-          auto matInverse = feat->Placement.getValue().toMatrix();
-          matInverse.inverse();
-          mat *= matInverse;
-      }
-      viewer->setupEditingRoot(csysDragger,&mat);
+    if (csysDragger && viewer) {
+        csysDragger->setUpAutoScale(viewer->getSoRenderManager()->getCamera());
+
+        auto originPlacement = App::GeoFeature::getGlobalPlacement(getObject()) * getObjectPlacement().inverse();
+        auto mat = originPlacement.toMatrix();
+
+        viewer->getDocument()->setEditingTransform(mat);
+        viewer->setupEditingRoot(csysDragger, &mat);
     }
 }
 
 void ViewProviderDragger::unsetEditViewer(Gui::View3DInventorViewer* viewer)
+{}
+
+void ViewProviderDragger::dragStartCallback(void* data, [[maybe_unused]] SoDragger* d)
 {
-    auto selection = static_cast<SoGroup*>(viewer->getSceneGraph());
-    SoNode *child = selection->getChild(0);
-    if (child && child->isOfType(SoPickStyle::getClassTypeId())) {
-        selection->removeChild(child);
-        viewer->setSelectionEnabled(true);
-    }
+    // This is called when a manipulator has done manipulating
+    auto sudoThis = static_cast<ViewProviderDragger*>(data);
```

Comment:
So, coin3D is from the time when C++ did not have lambdas so to implement closures context is passed via `void*` data argument that can contain arbitrary stuff and is defined when adding callback. 

In case of this callback `this` is passed as the data and later is extracted in context of the static function. So in reality this is just a `this` pointer passed explicitly to a function via argument. Some files uses `self` as name, in context of this file it was `sudoThis` - frankly I don't know why it was called like this, I assumed that it is convention that is used more across the codebase but it is not. 

So, any suggestions for the name? maybe `callerViewProvider`?